### PR TITLE
Make package/assembly/directory/project names match

### LINKED ISSUE #271: Assembly name of TestingHelpers
At least in the current `System.IO.Abstractions.TestingHelpers-2.1.0.178` the assembly itself is only named `TestingHelpers.dll`, i.e. no `System.IO.Abstractions` prefix. This is a inconsistency with the main `System.IO.Abstractions` and yells for name clashes: "TestingHelpers.dll" is quite generic.

Please consider naming the assembly `System.IO.Abstractions.TestingHelpers.dll` in future versions.

### PR DESCRIPTION
**Pull request by fgreinacher**
This PR makes package/assembly/directory/project names match.

Fixes #271

### CODE REVIEW

File: TestHelpers.Tests/TestHelpers.Tests.ncrunchproject [hunk 1]
```diff
@@ -1,20 +0,0 @@
-<ProjectConfiguration>
```
Comment (fgreinacher) [inline TestHelpers.Tests/TestHelpers.Tests.ncrunchproject]: @jpreese @tathamoddie Do you use NCrunch?

Comment (jpreese) [reply TestHelpers.Tests/TestHelpers.Tests.ncrunchproject]: I do, yes. Though I'm not convinced it needs to be in source control.

Comment (robkeim) [reply TestHelpers.Tests/TestHelpers.Tests.ncrunchproject]: If we're going to remove these I think it makes sense to update the `.gitignore` to ensure they don't accidentally work their way back into the repro.

It looks like we already have a section for NCrunch but I don't know why this line didn't cause these files to be ignored:
https://github.com/System-IO-Abstractions/System.IO.Abstractions/blob/e6ec3d48138fbf708c593953b3a395c23f28a84b/.gitignore#L75

Maybe that was updated after these files were already added?

Comment (fgreinacher) [reply TestHelpers.Tests/TestHelpers.Tests.ncrunchproject]: Ah, that’s why Git wants to remove them :)

**Comment by fgreinacher** [comment]
Any objections on merging this?

**Comment by jpreese** [comment]
This should be OK. As long as we're not making any changes to the namespace we shouldn't be breaking anything.

**Comment by jpreese** [comment]
I think when the conflicts are resolved we're all good 🎉

**Pull request merged**